src/server.c

```diff
@@ -3574,8 +3574,18 @@ void call(client *c, int flags) {
         /* Call propagate() only if at least one of AOF / replication
          * propagation is needed. Note that modules commands handle replication
          * in an explicit way, so we never replicate them automatically. */
-        if (propagate_flags != PROPAGATE_NONE && !(c->cmd->flags & CMD_MODULE))
+        if (propagate_flags != PROPAGATE_NONE && !(c->cmd->flags & CMD_MODULE)) {
+
+            /* Propagate a MULTI request once we encounter the first command which
+             * is a write command.
+             * This way we'll deliver the MULTI/..../EXEC block as a whole and
+             * both the AOF and the replication link will have the same consistency
+             * and atomicity guarantees. */
+            if (server.in_exec && !server.propagate_in_transaction)
+                execCommandPropagateMulti(c);
```

Comment:
what happens in this case:
```
MULTI
MODULE.WRITECMD key
EXEC
```
i think we will not propagate the MULTI...

Reply:
(because of the outer `if`)

Reply:
and a Q: flags with CMD_ADMIN never do dirty++, right?

Reply:
yeah, i suppose admin commands don't mess with the keyspace, except for DEBUG, which we don't care about here.

you're right about the modules, but what's the solution?
do i just replicate that new logic to before the other calls to `propagate` in this function?